tsdb/wal/wal.go

```diff
@@ -105,6 +105,15 @@ func (e *CorruptionErr) Error() string {
 	return fmt.Sprintf("corruption in segment %s at %d: %s", SegmentName(e.Dir, e.Segment), e.Offset, e.Err)
 }
 
+type listSegmentsErr struct {
```

Comment:
I don't think there is any benefit of an explicit error type here. We can include the directory error message at those 3 places in listSegments.

Reply:
Make sense.